### PR TITLE
Reject a media description if it's not bundled and policy is max-bundle.

### PR DESCRIPTION
**Pull request by taylor-b**
Also restructured the "rejecting media descriptions" sections since
there are now four things that can cause it to be rejected.

### CODE REVIEW

**Comment by pthatcherg** [comment]
I think this does a good job clarifying these edge cases and is the behavior we want.

File: draft-ietf-rtcweb-jsep.xml [hunk 1]
```diff
@@ -1102,7 +1102,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
                         bundle-only. This policy aims to minimize
                         candidate gathering and maximize multiplexing,
                         at the cost of less compatibility with legacy
-                        endpoints.</t>
+                        endpoints. When acting as answerer, the browser
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: typically we use 'endpoint' or 'implementation', rather than 'browser'

Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: I just copied the terminology used in the RTCP mux section. I'll change that as well.

File: draft-ietf-rtcweb-jsep.xml [hunk 2]
```diff
@@ -2667,21 +2669,26 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-                    indicated in <xref target="RFC3264"></xref>, Section
-                    6., and further processing for this m= section can
-                    be skipped.</t>
+                      <t>No supported codec is present in the offer.</t>
+
+                      <t>The bundle policy is "max-bundle" and the m=
+                      section is not in a BUNDLE group.</t>
+
+                      <t>The RTP/RTCP multiplexing policy is "require"
+                      and the m= section doesn't provide an
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: provide -> contain. Note that this text may need to be generalized if we go down the a=rtcp-mux-exclusive path.

File: draft-ietf-rtcweb-jsep.xml [hunk 3]
```diff
@@ -2667,21 +2669,26 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-                    is present in the offer, the corresponding
-                    m= section in the answer MUST be marked as rejected
-                    by setting the port in the m= line to zero, as
-                    indicated in <xref target="RFC3264"></xref>, Section
-                    6., and further processing for this m= section can
-                    be skipped.</t>
+                      <t>No supported codec is present in the offer.</t>
+
+                      <t>The bundle policy is "max-bundle" and the m=
+                      section is not in a BUNDLE group.</t>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: BUNDLE -> bundle, since it's not an acronym

File: draft-ietf-rtcweb-jsep.xml [hunk 4]
```diff
@@ -1102,7 +1102,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
                         bundle-only. This policy aims to minimize
                         candidate gathering and maximize multiplexing,
                         at the cost of less compatibility with legacy
-                        endpoints.</t>
+                        endpoints. When acting as answerer, the
+                        implementation will reject any m= section that
+                        is not in a BUNDLE group.</t>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: BUNDLE -> bundle

File: draft-ietf-rtcweb-jsep.xml [hunk 5]
```diff
@@ -1126,9 +1128,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
                         <t hangText="require:">The browser will only
                         gather RTP candidates. This halves the number
                         of candidates that the offerer needs to gather.
-                        When acting as answerer, the browser will
-                        reject any m= section that does not provide an
-                        "a=rtcp-mux" attribute.</t>
+                        When acting as answerer, the implementation
+                        will reject any m= section that does not
+                        provide an "a=rtcp-mux" attribute.</t>
```
Comment (juberti) [inline draft-ietf-rtcweb-jsep.xml]: provide -> contain

File: draft-ietf-rtcweb-jsep.xml (hunk 2 above)
Comment (taylor-b) [inline draft-ietf-rtcweb-jsep.xml]: For now I'm just restructuring the existing text here to make it more readable. I'll change "provide" to "contain" but we should probably do the generalizing it in a separate PR.

**Comment by juberti** [comment]
lgtm

File: draft-ietf-rtcweb-jsep.xml [hunk 6]
```diff
@@ -1126,9 +1128,9 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
                         <t hangText="require:">The browser will only
                         gather RTP candidates. This halves the number
                         of candidates that the offerer needs to gather.
-                        When acting as answerer, the browser will
-                        reject any m= section that does not provide an
-                        "a=rtcp-mux" attribute.</t>
+                        When acting as answerer, the implementation
+                        will reject any m= section that does not
+                        contain an "a=rtcp-mux" attribute.</t>
```
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: prefer that it not reject it but instead just treat it as if it does "a=rtcp-mux"

File: draft-ietf-rtcweb-jsep.xml [hunk 7]
```diff
@@ -2667,21 +2669,26 @@ candidate:1 1 UDP 1694498815 192.0.2.33 10000 typ host
...
-                    6., and further processing for this m= section can
-                    be skipped.</t>
+                      <t>No supported codec is present in the offer.</t>
+
+                      <t>The bundle policy is "max-bundle" and the m=
+                      section is not in a bundle group.</t>
+
+                      <t>The RTP/RTCP multiplexing policy is "require"
+                      and the m= section doesn't contain an
+                      "a=rtcp-mux" attribute.</t>
```
Comment (fluffy) [inline draft-ietf-rtcweb-jsep.xml]: same as above comment

**Comment by fluffy** [comment]
LGTM

**Comment by taylor-b** [comment]
I just realized this may not be complete. Even if max-bundle is used, shouldn't *one* media description be accepted? This would ensure the same result regardless of whether the max-bundle endpoint is the offerer or answerer. It would also prevent a remote offer from being rejected if it didn't include a "BUNDLE" group simply because there was only one media description.

**Comment by taylor-b** [comment]
Deficiencies should be addressed by this PR (apologies for missing them on the first pass): https://github.com/rtcweb-wg/jsep/pull/282

**Pull request merged**